core/editable.js

```diff
@@ -81,29 +96,72 @@
 					this.editor._.previousScrollTop = this.$.scrollTop;
 				}
 
-				// [IE] Use instead "setActive" method to focus the editable if it belongs to the host page document,
-				// to avoid bringing an unexpected scroll.
 				try {
-					if ( CKEDITOR.env.ie && !( CKEDITOR.env.edge && CKEDITOR.env.version > 14 ) && this.getDocument().equals( CKEDITOR.document ) ) {
-						this.$.setActive();
-					} else {
-						// We have no control over exactly what happens when the native `focus` method is called,
-						// so save the scroll position and restore it later.
-						if ( CKEDITOR.env.chrome ) {
-							var scrollPos = this.$.scrollTop;
-							this.$.focus();
-							this.$.scrollTop = scrollPos;
-						} else {
-							this.$.focus();
+					var isScrollLockFallbackNeeded = !CKEDITOR.env.safari && !CKEDITOR.env.chrome;
+
+					if ( isScrollLockFallbackNeeded ) {
+						var scrollables = [],
+							element = this;
+
+						if ( focusOptions && focusOptions.preventScroll ) {
+							while ( element ) {
+								if ( element.$.scrollHeight > element.$.clientHeight ) {
+									// Separate scopes for each variable, so it's accessible in listener.
+									( function() {
+										var scrollable = element,
+											scrollTop = element.$.scrollTop;
+
+										scrollables.push( {
+											element: scrollable,
+											scrollTop: scrollTop
+										} );
+
+										// IE and Edge scroll asynchronously.
+										if ( CKEDITOR.env.ie ) {
```

Comment:
I could drop IIFE by moving this part to separe forEach loop. But I'm not sure if it's better to loop two times, or just use same loop, but with IIFE for function scoping.

Reply:
I'm for keeping IIFE.